tests/unit/amo/components/TestRatingsByStar.js

```diff
@@ -148,30 +155,59 @@ describe(__filename, () => {
     const addon = addonForGrouping(grouping);
     store.dispatch(setGroupedRatings({ addonId: addon.id, grouping }));
     const root = render({ addon });
-    const counts = root.find('.RatingsByStar-star').find(Link);
-    const bars = root.find('.RatingsByStar-barContainer').find(Link);
 
-    function validateLink(link, score, expectedTitle) {
-      expect(link).toHaveProp(
-        'to',
-        reviewListURL({ addonSlug: addon.slug, score }),
+    const tableRow = root.find('.RatingByStar-table-row');
+
+    function validateLink(row, score, expectedTitle) {
+      const clientApp = CLIENT_APP_ANDROID;
```

Comment:
This test is using values that will be present because they are the default values, but we really shouldn't rely on those in a test. If we're going to have an assertion that includes `clientApp` and/or `lang`, then we should explicitly set those in the test, and the assertion should check for the values we set. You can see an example of how this can be done in the test for ['renders an opensearch link for Firefox' in TestSearchForm.js](https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestSearchForm.js#L179-L190).

Reply:
Thanks @bobsilverberg :) Actually, I was looking this type of approach only but could not able to find it. I will update this.

Reply:
@bobsilverberg I tried this approach but getting an error.
This is the code I implemented:
```
  it('adds a `src` query parameter to the review links when available in the location', () => {
    const src = 'some-src';
    const grouping = {
      5: 964,
      4: 821,
      3: 543,
      2: 22,
      1: 0,
    };
    const addon = addonForGrouping(grouping);
    const clientApp = CLIENT_APP_FIREFOX;
    const lang = 'fr';

    const customStore = dispatchClientMetadata({ clientApp, lang }).store;
    customStore.dispatch(setGroupedRatings({ addonId: addon.id, grouping }));

    const root = render({
      addon,
      location: createFakeLocation({ query: { src } }),
    });
    const tableRow = root.find('.RatingByStar-table-row');

    function validateLink(row, score) {
      row.simulate('click', createFakeEvent());

      sinon.assert.calledWith(
        fakeHistory.push,
        `/${lang}/${clientApp}${reviewListURL({
          addonSlug: addon.slug,
          score,
          src,
        })}`,
      );
    }

    validateLink(tableRow.at(0), '5');
    validateLink(tableRow.at(1), '4');
    validateLink(tableRow.at(2), '3');
    validateLink(tableRow.at(3), '2');
    validateLink(tableRow.at(4), '1');
  });
```

Here is the error:
```
 AssertError: expected stub to be called with arguments

      241 |       row.simulate('click', createFakeEvent());
      242 | 
    > 243 |       sinon.assert.calledWith(
          |                    ^
      244 |         fakeHistory.push,
      245 |         `/${lang}/${clientApp}${reviewListURL({
      246 |           addonSlug: addon.slug,

      at Object.fail (node_modules/sinon/lib/sinon/assert.js:107:21)
      at failAssertion (node_modules/sinon/lib/sinon/assert.js:66:16)
      at Object.calledWith (node_modules/sinon/lib/sinon/assert.js:92:13)
      at validateLink (tests/unit/amo/components/TestRatingsByStar.js:243:20)
      at Object.<anonymous> (tests/unit/amo/components/TestRatingsByStar.js:253:5)

```

Reply:
I did a really stupid mistake :sweat_smile: I solved it. I forgot to pass history and store to render method. Thanks :)